asset/asset.go

```diff
@@ -1045,3 +1053,64 @@ func (a *Asset) Leaf() (*mssmt.LeafNode, error) {
 	}
 	return mssmt.NewLeafNode(buf.Bytes(), a.Amount), nil
 }
+
+// ValidateAssetName validates an asset name (the asset's genesis tag).
+func ValidateAssetName(name string) error {
+	if len(name) == 0 {
+		return fmt.Errorf("asset name cannot be empty")
+	}
+	if len(name) > MaxAssetNameLength {
+		return fmt.Errorf("asset name cannot exceed %d bytes",
+			MaxAssetNameLength)
+	}
+	return nil
+}
+
+// escapeWhiteSpace escapes the given character if it is a white space
+// character.
+func escapeWhiteSpace(char int32) string {
```

Comment:
Should we use the internal `rune` type to make this a bit more explicit?

Reply:
Good point!